sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationClientCredential.java

```diff
@@ -53,14 +53,20 @@
     /**
      * Requires resource access key to create the credential
      * @param accessKey resource access key as provided by Azure in Base64 format
-     * @throws NoSuchAlgorithmException if HmacSHA256 is not available
-     * @throws InvalidKeyException if accessKey provided is not valid
      */
-    public CommunicationClientCredential(String accessKey) throws NoSuchAlgorithmException, InvalidKeyException {
+    public CommunicationClientCredential(String accessKey) {
         Objects.requireNonNull(accessKey, "'accessKey' cannot be null");
         byte[] key = Base64.getDecoder().decode(accessKey);
-        sha256HMAC = Mac.getInstance("HmacSHA256");
-        sha256HMAC.init(new SecretKeySpec(key, "HmacSHA256"));
+        Mac sha256HMAC = null;
+        try {
+            sha256HMAC = Mac.getInstance("HmacSHA256");
+            sha256HMAC.init(new SecretKeySpec(key, "HmacSHA256"));
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
```

Comment:
Let's talk about this a bit more. I wonder if somehow `sha256HMAC` instantiation and `init` should be moved to a builder mechanism to avoid throwing exceptions or maybe the first time that we call `appendAuthorizationHeaders` we set it up. @srnagar what do you think?

Reply:
We do use [builders for creating credentials](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/identity/azure-identity/src/main/java/com/azure/identity/ClientCertificateCredentialBuilder.java) in `azure-identity`. 

We also [use constructors](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/credential/AzureKeyCredential.java) as well for creating some credentials. So, both patterns are acceptable.

Reply:
Regarding the exception handling itself, if the `sha256HMAC` cannot be instantiated, we should throw this error but it should be a RuntimeException instead of a checked exception. Also, we should not use `e.printStackTrace()`. Log the and throw the exception instead using `ClientLogger`.

Reply:
Done, thanks Srikanta!